bids-validator/schema/datatypes/anat.yaml

```
@@ -0,0 +1,174 @@
+---
+# Nonparametric
+- suffixes:
+    - T1w
+    - T2w
+    - PDw
+    - T2starw
+    - FLAIR
+    - inplaneT1
+    - inplaneT2
+    - PDT2
+    - angio
+    - T2star  # deprecated
+    - FLASH  # deprecated
+    - PD  # deprecated
+  extensions:
+    - .nii.gz
+    - .nii
+    - .json
+  entities:
+    subject: required
+    session: optional
+    run: optional
+    acquisition: optional
+    ceagent: optional
+    reconstruction: optional
+    part: optional
+# Parametric
+- suffixes:
+    - T1map
+    - T2map
+    - T2starmap
+    - R1map
+    - R2map
+    - R2starmap
+    - PDmap
+    - MTRmap
+    - MTsat
+    - UNIT1
+    - T1rho
+    - MWFmap
+    - MTVmap
+    - PDT2map
+    - Chimap
+    - S0map
+    - M0map
+  extensions:
+    - .nii.gz
+    - .nii
+    - .json
+  entities:
+    subject: required
+    session: optional
+    run: optional
+    acquisition: optional
+    ceagent: optional
+    reconstruction: optional
+# Defacemask
+- suffixes:
+    - defacemask
+  extensions:
+    - .nii.gz
+    - .nii
+    - .json
+  entities:
+    subject: required
+    session: optional
+    run: optional
+    acquisition: optional
+    ceagent: optional
+    reconstruction: optional
+    modality: optional
```

Going by our current [regex](https://github.com/bids-standard/bids-validator/blob/fcc1c296c02d09c434aeea066b7d65e8836cc0c6/bids-validator/bids_validator/rules/file_level_rules.json#L49-L95), this should not have `ceagent`. Also, your regex construction seems to depend on the ordering of these entities, which may be why the test is failing.

```suggestion
    subject: required
    session: optional
    acquisition: optional
    reconstruction: optional
    run: optional
    modality: optional
```

---

Makes sense. The entity order should be inferred from the `entities.yaml` file, rather than the datatype files.

---

Thanks, fixing the ordering issue pointed out another test issue with modality vs mod (not reading the entity key from `entities.yaml`). That gets the anat example working as expected.

The eeg example taken from the existing tests `/sub-05/eeg/sub-05_task-matchingpennies_recording-eyetracking_physio.tsv.gz` looks like it wouldn't match the schema because a physio suffix isn't specified for eeg, does that seem accurate or I am misinterpreting it?

---

Physio is an odd one out, specification adds it too a bunch of modalities in its own section:
https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/06-physiological-and-other-continuous-recordings.html